src/parser.rs

```diff
@@ -149,7 +152,10 @@ where
         // FIXME(eddyb) implement error reporting support like in `input_consume_left`
         match self.state.forest.input(self.remaining).match_right(pat) {
             Some(n) => {
-                let (before, matching, _) = self.remaining.split_at(self.remaining.len() - n);
+                let (before, matching) = self
+                    .remaining
+                    .split_at::<Unknown>(/*self.remaining.len() - n*/ unimplemented!())
```

Comment:
Backing up indices safely is difficult. However, it is possible already in `indexing-str` if you have a container back-reference.